include/sympack/symPACKMatrix2D.hpp

```diff
@@ -897,16 +1077,59 @@ namespace symPACK{
           auto diag = dynamic_cast<const blockCell_t*>(pdiag);
 
           bassert(diag->nblocks()>0);
-          auto diag_nzval = diag->_nzval;
 
           auto snode_size = std::get<0>(_dims);
+          auto diag_nzval = diag->_nzval;
           auto nzblk_nzval = _nzval;
+#ifdef CUDA_MODE
+          if (diag->is_gpu_block || snode_size*total_rows() > symPACK::trsm_limit) {
+            
+            dev_ptr diag_nzval_inter, nzblk_nzval_inter;
+            nzblk_nzval_inter = symPACK::gpu_allocator.allocate<T>(snode_size*total_rows());                
+
+            if (diag->is_gpu_block) {
+                diag_nzval_inter = diag->_d_nzval;
+            } else {
+                diag_nzval_inter = symPACK::gpu_allocator.allocate<T>(snode_size*snode_size);
+            }
+            
+            if (diag_nzval_inter==nullptr || nzblk_nzval_inter==nullptr) {
+                blas::Trsm('L','U','T','N',snode_size, total_rows(), T(1.0),  diag_nzval, snode_size, nzblk_nzval, snode_size);
```

Comment:
IIUC, this branch handles the case where one or both of the calls to `symPACK::gpu_allocator.allocate` above returns `nullptr` (ie. due to device segment exhaustion).

However, for the case where one allocation succeeds and the other fails, I believe this control path results in _leaking_ one device allocation, which of course increases the chances that later allocations will also fail.